doc/source/dev/development_ghcodespaces.rst

```diff
@@ -0,0 +1,129 @@
+.. _development_ghcodespaces:
+
+
+Using GitHub Codespaces for NumPy development
+=============================================
+
+This section of the documentation will guide you through:
+
+*  using GitHub Codespaces for your NumPy development environment
+*  creating a personal fork of the NumPy repository on GitHub
+*  a quick tour of GitHub Codespaces and VSCode desktop application
+*  working on the NumPy documentation in GitHub Codespaces
+
+GitHub Codespaces
+-----------------
+`GitHub Codespaces`_  is  
+
+TO DO
+
+
+What is a codespace?
+--------------------
+A codespace is a development environment that is hosted in the cloud.
+Each codespace runs on a virtual machine hosted by GitHub. You can choose 
+the type of machine you want to use, depending on the resources you need. Various 
+types of machine are available, starting with a 2-core processor, 4 GB of RAM, 
+and 32 GB of storage.
+You can connect to a codespace from your browser, from Visual Studio Code, from 
+the JetBrains Gateway application, or by using GitHub CLI.
+
+
+Forking the NumPy repository
+----------------------------
+The best way to work on NumPy as a contributor is by making a fork of the 
+repository first.
+
+#. Browse to the `NumPy repository on GitHub`_ and `create your own fork`_.
+#. Browse to your fork. Your fork will have a URL like 
+   https://github.com/inessapawson/NumPy, except with your GitHub username in place of ``inessapawson``.
+   
+   
+Starting GitHub Codespaces
+--------------------------
+TO DO
+
+
+Quick workspace tour
+--------------------
+You can develop code in a codespace using your choice of tool:
+
+* a command shell, via an SSH connection initiated using GitHub CLI.
+* one of the JetBrains IDEs, via the JetBrains Gateway.
+* the Visual Studio Code desktop application.
+* a browser-based version of Visual Studio Code.
+
+In this quickstart, we will be using the VSCode desktop application as the editor. If you have not used it before, see the Getting started `VSCode docs`_ to familiarize yourself with this tool.
+
+Your workspace will look similar to the image below:
+
+TO DO
```

Comment:
Not sure how to best get images here...

<img width="1909" alt="cs-numpy" src="https://user-images.githubusercontent.com/9445180/205398681-9c555869-7bad-4a5d-835c-1ead8db1f8dc.png">

Reply:
I'll embed the image. Thank you for all your feedback, @craiglpeters!